index.js

```diff
@@ -573,8 +574,23 @@ class Raspi extends EventEmitter {
     throw new Error('reportDigitalPin is not yet implemented');
   }
 
-  pingRead() {
-    throw new Error('pingRead is not yet implemented');
+  pingRead(config, handler) {
+    const pulseOut = config.pulseOut;
+    const pin = config.pin;
+    const sonar = new Sonar(config.pin);
+    
+    const pinInstance = this[getPinInstance](this.normalize(pin));
+    if (pinInstance.mode != INPUT_MODE) {
+      this.pinMode(pin, INPUT_MODE);
```

Comment:
> I'm not sure about the below line though.

> > There's one catch though, we'll have to set the mode on the pinInstance to UNKNOWN_MODE though inside of the pinMode method, since Johnny-Five won't recognize PING_READ_MODE.

> Based on @rwaldron 's update to pingRead settings handler, it read "No pin mode specified". So my question is; do we still need to worry about the catch? And if so, do you mean something like the below? (sorry if i'm completely off track here).

Hmm...ok, so here's the issue I've been thinking about: the RPi is kinda unlike the Arduino in that the "specialty" pins (I2C, UART, SPI, etc) can also be used for GPIO, so our mode logic is a little trickier on the Pi. We have to always know what mode the pin is in so we can potentially switch it to something else later. In other words, we _have_ to have a way of determining what mode a pin is it (it always has a mode, logically speaking, even if we don't specify it on the `mode` property).

We can make the logic more complicated throughout raspi-io to support pingRead not specifying pinMode, but every mode check will get more complicated and I'd really prefer to avoid that.

@rwaldron does this sound ok setting the pin mode to 98?